### PR TITLE
general component

### PR DESCRIPTION
**Pull request by NRaeewal**

### CODE REVIEW

**Comment by NRaeewal** [comment]
[colorful block (general component) (text, button, img)](https://trello.com/c/zprEC7rf/28-colorful-block-general-component-text-button-img)

File: app/javascript/components/general/general_post.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  <div className="general-component-card fgs-al fgs-al-v fgs-al-g-30">
+    <div className="content-block fgs-al fgs-ali fgs-al-h fgs-al-g-60 fgs-al-align-items-center fgs-al-justify-content-space-evenly">
+      <div className={`image-block fgs-ali ${positionImageRight ? 'right' : ''}`}>
+        <img className="card-image" src={image} alt="general component" />
+      </div>
+
+      <div className="content fgs-al fgs-ali fgs-al-v fgs-al-g-30 ">
+        <div className="title fgs-ali heading-regular fgs-al-g-16">{header}</div>
+        <p className="text fgs-ali">{text}</p>
+        <a href={link}>
```
Comment (AnnaBaranova) [inline app/javascript/components/general/general_post.jsx:14]: @NRaeewal I think we don't need a link to wrap a button. What about passing onClick as a Prop to manage button functionality...? Also, I would pass onHide element to hide the button if we don't need it...

Comment (AnnaBaranova) [reply app/javascript/components/general/general_post.jsx:14]: @NRaeewal let's add a ButtonClass prop too to change button style if we want too...

File: app/javascript/components/general/general_post.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const GeneralPost = ({ image, header, text, buttonText, link, positionImageRight, buttonShow, buttonClass }) => {
+  const handleButtonClick = () => window.open(link);
```
Comment (AnnaBaranova) [inline app/javascript/components/general/general_post.jsx:5]: @NRaeewal I don't think this component should control onClick behavior. It should be a dummy styled component and any parent send props to use it. I would send  onClick. as a props Check a Link component you added to this project. the same thing. We can send onClick there and Link just runs it....

File: app/javascript/components/general/general_post.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    <div className="general-component-card fgs-al fgs-al-v fgs-al-g-30">
+      <div className="content-block fgs-al fgs-ali fgs-al-h fgs-al-g-60 fgs-al-align-items-center fgs-al-justify-content-space-evenly">
+        <div className={`image-block fgs-ali ${positionImageRight ? 'right' : ''}`}>
+          <img className="card-image" src={image} alt="general component" />
+        </div>
+
+        <div className="content fgs-al fgs-ali fgs-al-v fgs-al-g-30 ">
+          <div className="title fgs-ali heading-regular fgs-al-g-16">{header}</div>
+          <p className="text fgs-ali">{text}</p>
+          {buttonShow && (
```
Comment (AnnaBaranova) [inline app/javascript/components/general/general_post.jsx:17]: @NRaeewal  I think default behavior to show button, so I would do opposite. If I wanna special case, I send buttonHide. but if is's regular use case, I don't want to send extra props...

**Pull request merged**